.github/workflows/ci.yml

```diff
@@ -0,0 +1,27 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
+  pull_request: ~
+  workflow_dispatch: ~
```

Comment:
same haha, and yep your guess aligns with my understanding too. if we just had push without a branch specifier, it would run on push and when we create the PR and i figured it would be ok if we just ran it on just PR. similarly, i guess if we _need_ to ever force push to main, we could still have some CI. as for the workflow dispatch, i checked with the dev ex team and they've had some instances where they found it useful to be able to manually run a CI.